pkg/storage/bucket/swift/config.go

```diff
@@ -14,24 +14,27 @@ import (
 
 // Config holds the config options for Swift backend
 type Config struct {
-	AuthVersion       int            `yaml:"auth_version"`
-	AuthURL           string         `yaml:"auth_url"`
-	Username          string         `yaml:"username"`
-	UserDomainName    string         `yaml:"user_domain_name"`
-	UserDomainID      string         `yaml:"user_domain_id"`
-	UserID            string         `yaml:"user_id"`
-	Password          flagext.Secret `yaml:"password"`
-	DomainID          string         `yaml:"domain_id"`
-	DomainName        string         `yaml:"domain_name"`
-	ProjectID         string         `yaml:"project_id"`
-	ProjectName       string         `yaml:"project_name"`
-	ProjectDomainID   string         `yaml:"project_domain_id"`
-	ProjectDomainName string         `yaml:"project_domain_name"`
-	RegionName        string         `yaml:"region_name"`
-	ContainerName     string         `yaml:"container_name"`
-	MaxRetries        int            `yaml:"max_retries" category:"advanced"`
-	ConnectTimeout    time.Duration  `yaml:"connect_timeout" category:"advanced"`
-	RequestTimeout    time.Duration  `yaml:"request_timeout" category:"advanced"`
+	ApplicationCredentialID      string         `yaml:"application_credential_id"`
+	ApplicationCredentialName    string         `yaml:"application_credential_name"`
+	ApplicationCredentialSecret  string         `yaml:"application_credential_secret"`
```

Comment:
is the secret something actually secret that shouldn't be shared? it sounds like it. If yes, then can you use `flagext.Secret` like for the `Password` field in the same struct?

Reply:
Yes indeed, it should not be shared. changed.

Reply:
with this change the code doesn't compile again, can you take a look? https://github.com/grafana/mimir/actions/runs/10822125612/job/30046128413?pr=9181